src/v/pandaproxy/schema_registry/json.cc

```diff
@@ -630,6 +630,47 @@ bool is_object_superset(
       pj{newer})));
 }
 
+bool is_enum_superset(json::Value const& older, json::Value const& newer) {
+    auto older_it = older.FindMember("enum");
+    auto newer_it = newer.FindMember("enum");
+    auto older_is_enum = older_it != older.MemberEnd();
+    auto newer_is_enum = newer_it != newer.MemberEnd();
+
+    if (!older_is_enum && !newer_is_enum) {
+        // both are not an "enum" schema, compatible
+        return true;
+    }
+
+    if (!(older_is_enum && newer_is_enum)) {
+        // only one is an "enum" schema, not compatible
```

Comment:
How will this work if one is an `enum` and another is a `union` of `[null, enum]`?

I guess a separate check will iterate the union, and "deref" the enum and call this again?

Reply:
In principle yes. depending on the type of union ("anyOf" vs "oneOf") the element of the union is "dereferenced" for the check
so this should work (have to check)
```
reader = { "oneOf": [{ "enum" : [1, 2, 3] }, {"type": "null"}] }
writer = { "enum": [1, 3] }
```